google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb

```diff
@@ -0,0 +1,302 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Debugger
+      class Breakpoint
+        ##
+        # # Variable
+        #
+        # Represents a variable or an argument possibly of a compound object
+        # type. Note how the following variables are represented:
+        #
+        # A simple Variable:
+        # ```ruby
```

Comment:
I thought about using the `@example` tag here, but that requires adding the code to convert Ruby variables into `Google::Cloud::Debugger::Variable`, and use proper inspection format instead of just the simple JSON format. The additional verbosity makes the example less readable. So I decided to just keep it simple here to illustrate the relation between Ruby variable formats and `Google::Cloud::Debugger::Variable` formats.

I used the `@example` tag in the `Google::Cloud::Debugger::Variable.from_rb_var` method documentation to show the conversion details.